### DISCUSSION TITLE
Mount method with redirect doesn't work after 2.5.5 in certain situations

### DISCUSSION THREAD

**Discussion by rappasoft** [livewire/livewire#3773] [7. Bugs (archived)] [answered]
Consider this mount method in 2.5.5:

```php
public function mount()
{
    if (! auth()->user() || ! auth()->user()->isAdmin()) {
        return $this->redirectRoute('frontend.index');
    }
}
```

I am just doing an extra sanity/security check on the component (even though the page it's displayed on is behind middleware - in this case its a full page component) to make sure that if some malicious force was able to instantiate our component that it would not have access to the data that was inside.

Then the given test:

```php
public function admin_index_component_requires_an_authenticated_user()
{
    Livewire::test(Index::class)->assertRedirect('/');
}
```

In 2.5.5 this test passes, but in ^2.6 this test fails in the component view.

What's weird is if I dump right before the redirect, it dumps, so I know the redirect is being hit. But if I also dump in the render method it also gets hit in 2.6, but not 2.5.5

To make it even more strange, this only happens with full-page components. Using this mount method on stand-alone components (components rendered within a full-page component or regular blade view) the test passes.

This probably seems dumb and useless to most (it probably is) but before I remove it entirely, maybe it should be checked to see that it's not some sort of bug that no one else is encountering that is pointing to a bigger issue after the 2.6 release.

**Comment by joshhanley** [comment]
@rappasoft there was issues with query strings and URI's changing. Can you try and update to 2.6.3 and see if the issue is still happening?

I've just tested this locally with 2.6.3 and my test passed 🙂

Hope this helps!

**Comment by rappasoft** [comment]
@joshhanley It was strange, it passed as you said when I used a small component, but not if the component was being used as a full-page component.

Regardless, since this is a sanity check anyway I just ended up replacing the redirect with an abort_if. The changes of it getting hit are slim anyway.

Thanks for the help!

**Reply by joshhanley** [reply]
@rappasoft ah interesting! What does that bigger component do in the browser? Does it function/ redirect ok?

**Reply by rginnow** [reply]
I have a full page component that is very similar to the table from Caleb's Livewire screencasts. I have some redirect checks just like @rappasoft is talking about that would work fine before the big 2.6 update.

My program requires an access code and contract, which I store in the session. So in my `mount()` method, I do a simple check to see if the user has entered the info on the previous page.
```php
public function mount()
{
    if (!session()->has('access_code') || !session()->has('contract')) {
        return redirect()->route('access');
    }
}
```
However, the program seems to ignore it and go straight to the `render()` method. If I add a `dd("i made it here")` right after my if statement, it doesn't hit it. So it looks like the `return redirect()` is not actually returning a redirect, but just skipping straight to the render method.

Here is my render method:
```php
public function render()
{
    return view('livewire.student-list', [
        'user' => $this->user,
        'students' => $this->students // this is where it fails, because this custom property is not able to grab data
    ])->layout('layouts.app');
}
```

**Reply by rginnow** [reply]
Just verified to be sure... 2.5.5 is fine and completes the redirect. 2.6.0 and up does not.

**Comment by joshhanley** [comment]
@rappasoft and @rginnow yeah there is a breaking change in 2.6. We changed how the redirect works due to issues with browser cache, but I hadn't anticipated the scenario you have and as such has introduced a breaking change. I'm working on a fix for it.

Hope this helps!

**Comment by joshhanley** [comment, marked as answer]
@rappasoft and @rginnow I have submitted a PR #3971 with a fix for this.

Hope this helps!
